src/base/path.cpp

```diff
@@ -52,8 +52,35 @@ const int PATHLIST_TYPEID = qRegisterMetaType<PathList>("PathList");
 
 namespace
 {
+#if defined(Q_OS_WIN)
+    bool isQStrUNCPath(const QString &path)
+    {
+        // strict format of \\host\drive\folder with no forbidden char to avoid conflict with normal path
+        if (path.left(2) != uR"(\\)"_s)
+            return false;
+        const auto lastSlash = path.lastIndexOf(u'\\');
+        if ((lastSlash - 1) <= 2)
+            return false;
+        const QRegularExpression regex {u"[\\0-\\37:?\"*<>|:/]"_s}; // no drive letter allowed C:/
+        return !regex.match(path.data()).hasMatch();
+    }
+#endif
+
     QString cleanPath(const QString &path)
     {
+#if defined(Q_OS_WIN)
+        // QT's cleanPath does not support UNC path
+        if (isQStrUNCPath(path))
+        {
+            // only trim off trailing \ which is meaningless
+            int index = path.size();
+            while (path[index - 1] == u'\\')
+                --index;
+            if (index <= (path.size() - 1))
+                return path.left(index);
+            return path;
```

Comment:
I don't really like this way. All paths are `QDir::cleanPath` processed except UNC paths. This creates discrepancy behavior which might create unforeseen surprises in the future.

Reply:
imo it can be removed with the idea you posted of storing `\\host\share` root as a string, it also helps to handle extra slashes like `\\host\share\xxx\\\\yyy`